base/io/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+#![no_std]
+
+use codec::{Decode, Encode};
+use gstd::{prelude::*, ActorId};
+use scale_info::TypeInfo;
+use types::primitives::*;
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo)]
+pub struct Base {
+    /// Original creator of the Base.
+    pub issuer: ActorId,
+
+    /// Specifies how an NFT should be rendered, ie "svg".
+    pub base_type: String,
+
+    /// Provided ny user during Base creation.
+    pub symbol: String,
+
+    /// Parts that the base has.
+    /// Mapping from `PartId` to fixed or slot `Part`.
+    pub parts: BTreeMap<PartId, Part>,
+}
+
+#[derive(Debug, Clone, Encode, Decode, TypeInfo, PartialEq, Eq)]
+pub enum EquippableList {
+    All,
+    Custom(BTreeSet<CollectionAndToken>),
+}
+#[derive(Debug, Clone, Default, Encode, Decode, TypeInfo, Eq, PartialEq)]
```

Comment:
```suggestion
}

#[derive(Debug, Clone, Default, Encode, Decode, TypeInfo, Eq, PartialEq)]
```
There should be a space between entities for a better readability.